File: Problems/0500_0599/0504_Base_7/Project_CS/Base_7.cs
using System;
using System.Linq;

public class Solution
{
    public string ConvertToBase7(int num)
    {
        bool isNegative = false;

        if (num == 0)
            return "0";
        else if (num < 0)
        {
            isNegative = true;
            num = -num;
        }

        string resultStr = "";
        while (num > 0)
        {
            resultStr = (num % 7).ToString() + resultStr;
            num /= 7;
        }

        if (isNegative)
            return "-" + resultStr;
        else
            return resultStr;
    }

    public void Main(string args)
    {
        int num = int.Parse(args.Replace("[","").Replace("]","").Trim());

        Console.WriteLine("num = " + num.ToString());
        
        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();

        string result = ConvertToBase7(num);
        Console.WriteLine("result = " + result);
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}

File: Problems/0500_0599/0504_Base_7/Project_Python3/Base_7.py
# coding: utf-8

import os
import sys
import time

class Solution:
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        resultStr = ""

        if num == 0:
            return "0"
        elif num > 0:
            isNegative = False
        else:
            isNegative = True
            num = -num

        while num > 0:
            resultStr = str(num % 7) + resultStr
            num //= 7

        if isNegative:
            return "-" + resultStr
        else:
            return resultStr

def main():
    argv = sys.argv
    argc = len(argv)

    if argc < 2:
        print("Usage: python {0} <testdata.txt>".format(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("{0} not found...".format(argv[1]))
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        temp = temp.strip()
        if temp == "":
            continue
        print("args = {0}".format(temp))
        loop_main(temp)
    #   print("Hit Return to continue...")
    #   input()

def loop_main(temp):
    num = int(temp.replace("[","").replace("]","").rstrip())
    print("num = {0:d}".format(num))

    sl = Solution()
    time0 = time.time()
    result = sl.convertToBase7(num)

    print("result = {0}".format(result))

    time1 = time.time()
    print("Execute time ... : {0:f}[s]\n".format(time1 - time0))

if __name__ == "__main__":
    main()
